src/PHPePub/Core/EPub.php

```diff
@@ -666,27 +666,41 @@ function addFileToMETAINF($fileName, $fileData) {
     /**
      * Add dynamically generated data as a file to the book.
      *
-     * @param string $fileName Filename to use for the file, must be unique for the book.
+     * @param string $filePath File panh and name to use for the file, must be unique for the book.
      * @param string $fileId   Unique identifier for the file.
      * @param string $fileData File data
      * @param string $mimetype file mime type
      *
      * @return bool $success
      */
-    function addFile($fileName, $fileId, $fileData, $mimetype) {
-        if ($this->isFinalized || array_key_exists($fileName, $this->fileList)) {
+    function addFile($filePath, $fileId, $fileData, $mimetype) {
+        if ($this->isFinalized || array_key_exists($filePath, $this->fileList)) {
             return false;
         }
         if (!$this->isInitialized) {
             $this->initialize();
         }
-        $fileName = FileHelper::normalizeFileName($fileName);
+        $filePath = FileHelper::normalizeFileName($filePath);
+
+        $components = explode('/', $filePath);
+        $lastComponent = end($components);
+
+        // Ensure the length of file names is within the spec.
+        // https://www.w3.org/TR/epub/#sec-container-filenames
+        assert(strlen($lastComponent) <= 255, "File name greater than 255 characters: " . $lastComponent);
+        assert(strlen($filePath) <= 65535, "File path greater than 65535 characters: " . $filePath);
```

Comment:
Even if someone enables assertions in "production" this seems safe to me because the spec says this specifically.